ompi/mpi/java/java/Request.java

```diff
@@ -113,6 +117,22 @@ public final void cancel() throws MPIException
 	private native void cancel(long request) throws MPIException;
 
 	/**
+	 * Adds a buffer to the buffer array list.  This method should
+	 * be called by the internal api whenever a persistent request
+	 * is created and any time a request object, that has an associated
+	 * buffer, is returned from a non-blocking operation to protect 
+	 * the buffers from being garbage collected prematurely.
+	 * @param buf buffer to add to the array list
+	 */
+	protected final void addBufRef(Buffer buf)
+	{
+		if(this.buffers == null)
```

Comment:
El 19/08/15 a las 19:20, Nathaniel Graham escribió:
>
> In ompi/mpi/java/java/Request.java 
> <https://github.com/open-mpi/ompi/pull/820#discussion_r37441459>:
>
> > @@ -113,6 +117,22 @@ public final void cancel() throws MPIException
> >  	private native void cancel(long request) throws MPIException;
> >
> >  	/**
> > +	 * Adds a buffer to the buffer array list.  This method should
> > +	 * be called by the internal api whenever a persistent request
> > +	 * is created and any time a request object, that has an associated
> > +	 * buffer, is returned from a non-blocking operation to protect
> > +	 * the buffers from being garbage collected prematurely.
> > +	 * @param buf buffer to add to the array list
> > +	 */
> > +	protected final void addBufRef(Buffer buf)
> > +	{
> > +		if(this.buffers == null)
>
> The 8 space tabs are now present in all of the java code. I am not 
> sure what is common in the java world, but I believe it is one of 
> those little issues that people debate back and forth quite a bit. 
> Based on a quick google, it looks like maybe I cannot make eclipse do 
> 4 space tabs. I can play with it if you like.
>

You can make eclipse do 4 space tabs:

Window / Preferences / General / Editors / Text Editors:
     Displayed tab width: 4
     Insert spaces for tabs: yes

> Ill keep what you said about the if statements in mind, but in my 
> updated fix I am currently working on, I have removed the if statement.
>
> —
> Reply to this email directly or view it on GitHub 
> <https://github.com/open-mpi/ompi/pull/820/files#r37441459>.
>